### ISSUE TITLE
Generic.ControlStructures.InlineControlStructure confused by mixed short/long tags

### ISSUE THREAD

**Issue by igorsantos07** [squizlabs/PHP_CodeSniffer#2565]
Example:
```php
<?php if (true) : ?>
<? endif ?>
<?php if (true) { ?>
<? } ?>

<? if (false) : ?>
<? endif ?>
<?php if (false) { ?>
<?php } ?>
```

CLI output (reduced for brevity):
```
$ phpcs -n --standard=PSR2 phpcs-test.php
  1 | ERROR | [x] Inline control structures are not allowed
  3 | ERROR | [x] Inline control structures are not allowed
$ phpcbf -n --standard=PSR2 phpcs-test.php
  /home/me/phpcs-test.php                            FAILED TO FIX
```

Resulting "fixed" file:
```php
<?php if (true) {
    :;
} ?>
<? endif ?>
<?php if (true) { { { { { { { { { { { { { { { { { { { { { { { { { { { {;}}}}}}}}}}}}}}}}}}}}}}}}}} } ?>
<? } ?>

<? if (false) : ?>
<? endif ?>
<?php if (false) { ?>
<?php } ?>
```

As can be seen in the `false` if's, using `<?php` or `<?` on both lines do not trigger the phpcs error. I'm not so sure what `InlineControlStructure` means in those cases (maybe the humanized message doesn't make much sense for those cases?), but we can say for sure that `phpcbf` got lost with the alt-syntax and nuts with the bracketed one :joy: 

That's for versions 3.3.2/3.4.2 at PHP 7.2.20.

**Comment by gsherwood** [comment]
This is probably due to that second PHP block using short open tags. PHPCS is probably being told it is inline HTML by the PHP tokenizer, so the closing tag of the IF statement is never found. So to PHPCS it looks like a syntax error.

**Comment by igorsantos07** [comment]
I've updated the description to reflect new findings. Nonetheless, that's probably not related to the short tags, at least not in the way you mentioned. PHPCS has sniffers for that, and if you remove `-n` you'll see it's also emitting a warning about that; I just added `-n` as the warning is not relevant to the bug - in our project, short tags are allowed just fine and the issue also happens.

**Comment by igorsantos07** [comment]
Updated to 3.4.2 and the issue is exactly the same.

**Comment by jrfnl** [comment]
> Nonetheless, that's probably not related to the short tags

@igorsantos07 Actually it is. Sniffs work independently of each other and as you will see if you use `-s` as well, the other error is from another sniff.

The system on which you are running PHPCS must be configured with `short_open_tag=Off` for them not to be recognized by the `InlineControlStructure` sniff.
The `ShortOpenTags` sniff makes a special allowance for that to still be able to detect them and warn about them appropriately.

I think the bug fix which is needed here, is for the `InlineControlStructure` sniff to still throw an error when it detects this, but not try to auto-fix.

**Comment by igorsantos07** [comment]
Correct, sometime later I noticed another issue, but that time with an entire class file and phpcs reported it could be a `short_open_tag` issue as it found no code in that file. I had upgraded recently to PHP 7.2 and my ini file went back to the defaults :man_facepalming: 

I'm not sure what to say here. How should phpcs behave when the code is incomplete? I guess we can simply ignore the fact it was complaining about something weird (inline control structures are forbidden), because in fact the rest of the code was "missing"?
Also, should it be able to double check cases like and maybe warn about the missing config, or the actual hint I got in the full-class file was just a cautionary message with no extra checks? It took me a while to figure out what was going on, as phpcs was running locally but the code is executed in another server (which was properly setup for `<?`).

Not to mention it's definitely "funny" what it did with the `if (true) {` haha

**Comment by jrfnl** [comment]
> How should phpcs behave when the code is incomplete?

Generally speaking, PHPCS will bow out and not report anything or, at the very least, not try to auto-fix code if it doesn't find the tokens which are expected, as this normally indicates that PHPCS is run during _live coding_ or that there is a parse error.

That way the fixer won't accidentally introduce (additional) syntax errors.

> Not to mention it's definitely "funny" what it did with the if (true) { haha

`True` that :joy:

**Comment by jrfnl** [comment]
@igorsantos07 PR #2567 should fix this by bowing out in the precise situation you found yourself in. Hope it helps ;-)
